cpp/src/Ice/ConnectionI.cpp

```diff
@@ -49,6 +50,25 @@ namespace
         Ice::ConnectionI* _connection;
     };
 
+    class KeepAlive final : public IceUtil::TimerTask
+    {
+    public:
+        KeepAlive(const Ice::ConnectionIPtr& connection) : _connection(connection) {}
+
+        void runTimerTask() final
+        {
+            if (auto connection = _connection.lock())
+            {
+                // TODO: handle synchronous and asynchronous failures
```

Comment:
IceRPC's version:
https://github.com/icerpc/icerpc-csharp/blob/57995b702436169798fc3fe5fd513e6f09591397/src/IceRpc/Internal/IceProtocolConnection.cs#L630

suggests we should do better here. In particular, don't send a heartbeat when one is already outstanding or has failed.